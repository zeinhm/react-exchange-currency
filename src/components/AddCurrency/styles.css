.add-currency {
  display: flex;
  justify-content: center;
  margin: 1rem;
}

.add-currency select {
  width: 100%;
}