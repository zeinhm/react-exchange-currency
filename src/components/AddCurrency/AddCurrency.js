import React from 'react'
import { SUPPORTED_CURRENCIES } from '../../lib/constants'
import './styles.css'

export default function AddCurrency(props) {
  const { onSubmit, selectedCurrency, setSelectedCurrency } = props
  return (
    <form className='add-currency' onSubmit={onSubmit}>
      <select
        value={selectedCurrency}
        onChange={e => setSelectedCurrency(e.target.value)}
      >
        <option value="" disabled defaultValue>(+) Add More Currencies</option>
        {SUPPORTED_CURRENCIES.map(option => (
          <option key={option} value={option}>{option}</option>
        ))}
      </select>
      {selectedCurrency && <input type="submit"/>}
    </form>
  )
}
