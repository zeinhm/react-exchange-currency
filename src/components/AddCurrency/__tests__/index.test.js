import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import AddCurrency from '../AddCurrency';

describe('src/components/AddCurrency', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const props = {
      currencies: ['test'],
      onSubmit: () => {},
      selectedCurrency: 'test',
      setSelectedCurrency: () => {}
    }
    const tree = shallow.render(<AddCurrency {...props} />);
    tree.props.onSubmit();
    tree.props.children[0].props.onChange({ target: { value: 'test' } });
    expect(tree).toMatchSnapshot();
  });
});
