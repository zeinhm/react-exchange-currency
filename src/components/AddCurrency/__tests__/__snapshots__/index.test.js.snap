// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/AddCurrency render 1`] = `
<form
  className="add-currency"
  onSubmit={[Function]}
>
  <select
    onChange={[Function]}
    value="test"
  >
    <option
      defaultValue={true}
      disabled={true}
      value=""
    >
      (+) Add More Currencies
    </option>
    <option
      value="USD"
    >
      USD
    </option>
    <option
      value="CAD"
    >
      CAD
    </option>
    <option
      value="IDR"
    >
      IDR
    </option>
    <option
      value="GBP"
    >
      GBP
    </option>
    <option
      value="CHF"
    >
      CHF
    </option>
    <option
      value="SGD"
    >
      SGD
    </option>
    <option
      value="INR"
    >
      INR
    </option>
    <option
      value="MYR"
    >
      MYR
    </option>
    <option
      value="JPY"
    >
      JPY
    </option>
    <option
      value="KRW"
    >
      KRW
    </option>
  </select>
  <input
    type="submit"
  />
</form>
`;
